GitTfs.VsCommon/Wrappers.cs

```diff
@@ -527,7 +527,14 @@ public void Merge(string sourceTfsPath, string targetTfsPath)
             if (author != null)
                 checkinParameters.Author = author;
 
-            return _workspace.CheckIn(checkinParameters);
+            try
+            {
+                return _workspace.CheckIn(checkinParameters);
+            }
+            catch (GatedCheckinException gatedException)
+            {
+                throw new GitTfsGatedCheckinException(gatedException.ShelvesetName.Split(new[] { ';' })[0], gatedException.AffectedBuildDefinitions, gatedException.CheckInTicket);
```

Comment:
> `gatedException.ShelvesetName.Split(new[] { ';' })[0]`

Will it work with `gatedException.ShelvesetName`? It'd be nice to avoid needing to mess with the value if git-tfs doesn't need to.